src/Symfony/Component/Intl/NumberFormatter/NumberFormatter.php

```diff
@@ -520,10 +520,15 @@ public function parse($value, $type = self::TYPE_DOUBLE, &$position = 0)
             return false;
         }
 
-        preg_match('/^([^0-9\-\.]{0,})(.*)/', $value, $matches);
-
         // Any string before the numeric value causes error in the parsing
-        if (isset($matches[1]) && !empty($matches[1])) {
+        if (preg_match('/^-?\.?[0-9]++(?:,[0-9]++)*+(\.[0-9]++)?/', $value, $matches)) {
```

Comment:
This is not as strict as the native intl NumberFormatter, but it is a stricter enough to make our tests green when intl is disabled.